packages/loopring_v3/contracts/core/impl/ExchangeV3.sol

```diff
@@ -43,6 +43,9 @@ contract ExchangeV3 is IExchangeV3, ReentrancyGuard
     using ExchangeWithdrawals   for ExchangeData.State;
 
     ExchangeData.State private state;
+    address public  __DEPRECATED__loopringAddr; // kept for proxy compatibility
+    uint8   private __DEPRECATED__ammFeeBips;   // kept for proxy compatibility
+    bool    public  __DEPRECATED__allowOnchainTransferFrom; // kept for proxy compatibility
 
```

Comment:
I added these values to the struct, and that way they should still map to the correct storage slot in the old contract:
https://docs.soliditylang.org/en/v0.8.3/internals/layout_in_storage.html

> The elements of structs and arrays are stored after each other, just as if they were given as individual values.

I did some external testing with this behavior and I think I did it right.

Reply:
That's true if we never add variables to this contract. But if we do in the future, the new variable will conflict with these three deprecated variables we delete.

Reply:
The ones in the struct now map to the old variables outside of the struct. If variables are part of a struct or not is irrelevant for the storage slot the are assigned.

This

```
Struct Data
{
    uint dataA;
    uint dataB;
}

contract Ex
{
    Data   data;
    uint   dataC;
    uint   dataD;
    uint   dataE;
}
```

has the same storage layout as this:
```
Struct Data
{
    uint dataA;
    uint dataB;
    uint dataC;
    uint dataD;
}

contract Ex
{
    Data   data;
    uint   dataE;
}
```

So adding additional storage values after the struct creates new variables taking up new slots. The separate values in the old contract will map to the values in the struct in the new implemenation.

Reply:
Thank you for this knowledge which is good to know. I'll revert the change.